FEATURE_PROCESS.adoc

```diff
@@ -1,5 +1,106 @@
+* Draft a minimal feature analysis, based on the current https://github.com/wildfly/wildfly-proposals/blob/main/design-doc-template.adoc[design-doc-template]. This should include the expected stability level, links to the issues above, and enough Overview and User Story information to give people interested in being part of the feature team a sense of what would be involved.
+* Submit a draft analysis PR to the https://github.com/wildfly/wildfly-proposals[`wildfly-proposals`] repository.
+* Send an email to the `wildfly-dev` mail list telling people about the upcoming work and asking for volunteers to participate in the feature team.
+
+When the above tasks are complete, change the `Status` field of the WildFly Feature Planning project JIRA to `Planning`.
+
+=== Planning
+
+The Planning stage of a feature consists of <<feature-team,feature team>> formation and enough discussion of the feature among the team that they can agree on a target feature stability level and target WildFly release in which they expect to complete the feature.
+
+NOTE: The target WildFly release is just a target and is not a requirement.
+
+Update the feature analysis document with the planned stability level (if it has changed) and the GitHub ids of the feature team members.
+
+Set the `Iteration` field to the iteration for the WildFly release.
+
+When the above tasks are complete, change the `Status` field of the WildFly Feature Planning project JIRA to `In Progress`.
+
+IMPORTANT: Changes to the `Iteration` field of a WildFly Feature Planning project issue should always reflect the consensus of the feature team.
+
+=== In Progress
+
+The main work on the feature happens in this stage. The feature team works to meet the <<requirements>> that apply to the feature's target stability.
+
+When the team believes all development work on the feature is complete, change the `Status` field of the WildFly Feature Planning project JIRA to `Review`.
+
+NOTE: Developers should not move an issue to `Review` status as a means of asking other team members to do routine code review. Such intra-team coordination should be handled in some other way. Moving an issue to `Review` is a signal to WildFly release coordinators that work on the feature is about to conclude.
+
+=== Review
+
+In the Review stage the team formally validates that the <<requirements>> have been met.
+
+This should be a relatively quick stage, as the team should have a belief that the requirements are met before the feature issue is moved to `Review`. The work in this stage should simply be a formal check and signoff.
+
+When the signoff is complete, the feature team should notify the release coordinator for the current WildFly release.
```

Comment:
I wonder if that could be the time to use the assignee field?  I don't find pings / notifications scale well nowadays but reviewing which issues I am assigned gives me something I can periodically look at..

Reply:
Maybe we need to combine the naming discussion with the previous comment, I think ready for merge would be good so periodically we can look at what is ready and get it merged - but do we want a specific state for "Signoff" / "Reiew" or as that discussion is within the feature team could that be the last thing they do whilst in "In Progress"?

Reply:
I think you get a big vote on that one, as you'd be a prime beneficiary of the info provided by separating 'Review/Signoff' from 'In Progress'.

The other factor is we talked about a transition to 'Review' triggering automation to add a stability-specific checklist to the issue description, a la what Jakarta has in some of their issues. We could do that when the issue moves into In Progress, but then if the team changes the stability target things get messy.

Reply:
See above; for now we decided to drop the 'Review/Signoff' stage unless their is automation to bring value to it.